packages/wdio-runner/src/index.js

```diff
@@ -68,15 +68,15 @@ export default class Runner extends EventEmitter {
         const isMultiremote = this.isMultiremote = !Array.isArray(this.configParser.getCapabilities())
 
         /**
-         * create `browser` stub only if `specFiltering` feature is enabled
+         * create `browser` stub
          */
-        let browser = this.config.featureFlags.specFiltering === true ? await this._startSession({
+        let browser = await this._startSession({
```

Comment:
I don't quite understand, are things not dependent on the `specFiltering` flag anymore?

Reply:
I screwed up with this:
1. init reporter, session capabilities are not available at this point
2. init test framework
3. then we can create browser session
4. pass capabilities to reporter

before it was:
1. create browser session
2. init reporter
3. run framework

Reply:
to deal with it there are 3 ways in mind
- current approach, the easiest. Create browser stub always and pass capabilities to reporter so that it can create log file with capabilities.browserName or whatever
- pass user defined capabilities to reporter
- change test framework init somehow so that reporter can be initialised after test framework. I'm not sure if it's possible at all

Reply:
I've chosen to always create browser stub. I takes 5ms to create it on my macbook 2015.

We can also just pass user defined capabilities instead, I think it should work in most of the cases

Reply:
```
// do not change anything here
let browser = this.config.featureFlags.specFiltering === true ? await this._startSession({
    ...this.config,
    _automationProtocol: this.config.automationProtocol,
    automationProtocol: './protocol-stub'
}, caps) : undefined

this.reporter = new BaseReporter(this.config, this.cid, caps) // user defined caps
```

Reply:
@christian-bromann I've reverted the change and switch to approach with user defined caps. If you like the previous approach more it's easy to change it back.

Added unit tests!